### PR TITLE
(CLOUD-226 to CLOUD-228) Tests for ec2_vpc types

### PR DESCRIPTION
**Pull request by zreichert**
This PR covers
CLOUD-226
CLOUD-227
CLOUD-228

### CODE REVIEW

File: spec/acceptance/vpc_spec.rb [hunk 1]
```diff
@@ -219,27 +219,544 @@ def generate_ip
...
+
+      it 'deletes the ec2_vpc_internet_gateway' do
         ENV['AWS_REGION'] = @default_region
-        options = {:name => "#{@name}-vpc"}
-        @result = TestExecutor.puppet_resource('ec2_vpc', options, '--modulepath ../')
+        name = "#{@delete_me[:name]}-igw"
+        options = {:name => name, :region => @default_region, :ensure => 'absent'}
+        result = TestExecutor.puppet_resource('ec2_vpc_internet_gateway', options, '--modulepath ../')
+        expect(result.stderr).not_to match(/Error/)
+        exspect(find_internet_gateway(name)).to raise_error
```
Comment (garethr) [inline spec/acceptance/vpc_spec.rb]: Typo, presumably should be `expect`

File: spec/acceptance/vpc_spec.rb [hunk 2]
```diff
@@ -219,27 +219,544 @@ def generate_ip
...
+
+    end
+
+    context 'delete with puppet resource' do
+
       before(:all) do
+        @delete_me = @config.clone
+        @delete_me[:name] = "#{PuppetManifest.env_id}-#{SecureRandom.uuid}"
+        result = PuppetManifest.new(@template, @delete_me).apply
+        require 'pry'; binding.pry
```
Comment (garethr) [inline spec/acceptance/vpc_spec.rb]: Left in Pry call

File: spec/acceptance/vpc_spec.rb [hunk 3]
```diff
@@ -219,27 +219,544 @@ def generate_ip
       expect(@aws.tag_difference(vpc, new_config[:tags])).to be_empty
     end
 
-    describe 'using puppet resource on the VPC' do
+  end
+
+  describe 'createing a new VPC environment with all possible properties' do
```
Comment (garethr) [inline spec/acceptance/vpc_spec.rb]: Typo: createing rather than creating

**Comment by garethr** [comment]
@zreichert I've left a few small comments. Also, looking at the tests in `delete with puppet resource`, do these have an implied order in which they need to happen? For instance if I ran the delete route table test in isolation would it fail because the gateway it's attached to is still present? The other VPC tests work around this by using the relationships in the _delete manifest template.

**Comment by zreichert** [comment]
@garethr Correct there is an implied order.  I could just merger these all in to one giant it block.  This would solve the ordering problem but the report would not be as clean.

**Comment by zreichert** [comment]
@garethr Tests fixed.  Please review again.

**Pull request merged**